src/main/java/com/google/sps/servlets/GmailServlet.java

```diff
@@ -18,38 +18,34 @@
 import com.google.api.services.gmail.Gmail;
 import com.google.api.services.gmail.model.Message;
 import com.google.appengine.repackaged.com.google.gson.Gson;
-import com.google.sps.utility.AuthenticationUtility;
+import com.google.sps.model.AuthenticatedHttpServlet;
 import com.google.sps.utility.GmailUtility;
 import java.io.IOException;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /**
  * Serves selected information from the User's Gmail Account. TODO: Create Servlet Utility to handle
- * common functions (Issue #26) TODO: Create abstract class to handle authentication with each
- * request (Issue #37)
+ * common functions (Issue #26)
  */
 @WebServlet("/gmail")
-public class GmailServlet extends HttpServlet {
-
+public class GmailServlet extends AuthenticatedHttpServlet {
   /**
    * Returns messageIds from the user's Gmail account
    *
    * @param request Http request from the client. Should contain idToken and accessToken
    * @param response 403 if user is not authenticated, list of messageIds otherwise
+   * @param googleCredential valid google credential object (already verified)
    * @throws IOException if an issue arises while processing the request
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // get Google credential object. Ensure it is valid - otherwise return an error to client
-    Credential googleCredential = AuthenticationUtility.getGoogleCredential(request);
-    if (googleCredential == null) {
-      response.sendError(403, AuthenticationUtility.ERROR_403);
-      return;
-    }
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    // Null credentials (i.e. unauthenticated requests) should be handled by parent class
+    assert googleCredential != null;
```

Comment:
Nit: Turn the comment into an assertion message, e.g.:
`assert x > 0 : "x must be greater than zero"`

Reply:
Cool - didn't know this existed!